numpy/lib/function_base.py

```diff
@@ -514,7 +514,10 @@ def average(a, axis=None, weights=None, returned=False):
         avg = a.mean(axis)
         scl = avg.dtype.type(a.size/avg.size)
     else:
-        a = a + 0.0
+        # Cast bool, unsigned int, and int to float64 by default
```

Comment:
Someone else used a function that might be good here, `result_type`. So maybe something like
```
dt = np.result_type(a, 0.0)
```